scripts/timelapse-colorizer-data/data_writer_utils.py

```diff
@@ -101,6 +101,37 @@ def remap_segmented_image(
     return (seg_remapped, lut)
 
 
+def update_collection(collection_filepath, dataset_name, dataset_path):
+    """
+    Adds a dataset to a collection file, creating the collection file if it doesn't already exist.
+    If the dataset is already in the collection, the existing dataset path will be updated.
+    """
+    collection = []
+
+    # Read in the existing collection, if it exists
+    if os.path.exists(collection_filepath):
+        try:
+            with open(collection_filepath, "r") as f:
+                collection = json.load(f)
+        except:
+            collection = []
```

Comment:
If the collections file is malformed (such as a `.json` file that doesn't have anything in it), it will overwrite the collection file here. This happened once during my testing (I think I opened a file but didn't dump the JSON data), and I didn't want to regenerate the entire dataset again.